### ISSUE TITLE
T-SQL: Do not parse APPLY clauses as JOINs

### ISSUE THREAD

**Issue by borchero** [sqlfluff/sqlfluff#3688]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

When using `APPLY` clauses, subqueries cannot be prevented. However, SQLFluff interprets `APPLY` clauses similar to joins and, hence, applies rule L042 (`Join/From clauses should not contain subqueries. Use CTEs instead.`). This issue may be related to #3685.

### Expected Behaviour

The following query should not result in any linting issues even if `forbid_subquery_in = both` for rule L042:

```sql
SELECT table1.*
FROM table1
OUTER APPLY (
    SELECT *
    FROM table2
    WHERE table1.col = table2.col
) AS applied
```

### Observed Behaviour

SQLFluff tries to wrongly move the subquery to a CTE, resulting in an actual error:

```sql
WITH applied AS (
    SELECT *
    FROM table2
    WHERE table1.col = table2.col -- Reference 'table1.col' refers to table/view not found in the FROM clause or found in ancestor statement.
)
SELECT table1.*
FROM table1
OUTER APPLY applied

```

### How to reproduce

See query above.

### Dialect

T-SQL

### Version

`1.2.1` (homebrew)

### Configuration

```
[sqlfluff]
dialect = tsql
exclude_rules = L044

[sqlfluff:rules:L010]
capitalisation_policy = upper

[sqlfluff:rules:L030]
extended_capitalisation_policy = lower
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by fmms** [comment]
@borchero i believe this is a duplicate of #3572

**Comment by borchero** [comment]
@fmms yes, thanks!
